pkg/controller/envoy/ads_event.go

```diff
@@ -201,19 +213,23 @@ func (svc *ServiceEvent) handleLdsResponse(rsp *service_discovery_v3.DiscoveryRe
 		listener = &config_listener_v3.Listener{}
 	)
 	current := sets.New[string]()
+	lastRouteNames := svc.DynamicLoader.routeNames
+	svc.DynamicLoader.routeNames = []string{}
 	for _, resource := range rsp.GetResources() {
 		if err = anypb.UnmarshalTo(resource, listener, proto.UnmarshalOptions{}); err != nil {
 			continue
 		}
 		current.Insert(listener.GetName())
+		apiStatus := core_v2.ApiStatus_UPDATE
 		newHash := hash.Sum64String(resource.String())
 		if newHash != svc.DynamicLoader.ListenerCache.GetLdsHash(listener.GetName()) {
 			svc.DynamicLoader.ListenerCache.AddOrUpdateLdsHash(listener.GetName(), newHash)
 			log.Debugf("[CreateApiListenerByLds] update %s", listener.GetName())
-			svc.DynamicLoader.CreateApiListenerByLds(core_v2.ApiStatus_UPDATE, listener)
```

Comment:
why change it, let's keep consistent with cds

Reply:
because we rely on this interface（CreateApiListenerByLds） to resolve the internal routename.